src/corelibs/U2Core/src/dbi/DbiDocumentFormat.cpp

```diff
@@ -80,17 +80,19 @@ Document *DbiDocumentFormat::loadDocument(IOAdapter *io, const U2DbiRef &dstDbiR
 
     objects << prepareObjects(handle, objectIds);
 
+    U2DbiRef resultDbiRef = dstDbiRef;
     if (fs.value(DEEP_COPY_OBJECT, false).toBool()) {
         QList<GObject *> clonedObjects = cloneObjects(objects, dstDbiRef, fs, os);
         qDeleteAll(objects);
         CHECK_OP_EXT(os, qDeleteAll(clonedObjects), NULL);
         objects = clonedObjects;
     } else {
+        resultDbiRef = srcDbiRef;
```

Comment:
This is a wrong solution that breaks the method contract: the method must return document with dstDbiRef

The correct solution is to resolve the loaded object by the original reference:

MaEditorTasks.cpp:103
` GObject *obj = GObjectUtils::selectObjectByReference(unloadedReference, UOF_LoadedOnly);`